components/TextInput.js

```diff
@@ -1,28 +1,69 @@
-import PropTypes from 'prop-types';
 import React, { PureComponent } from 'react';
+import autoBindReact from 'auto-bind/react';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
 import { TextInput as RNTextInput } from 'react-native';
 
-import { connectStyle } from '@shoutem/theme';
 import { connectAnimation } from '@shoutem/animation';
+import { connectStyle } from '@shoutem/theme';
+
+import { Caption } from './Text';
+import { View } from './View';
+
+// Style properties defined in theme which would case a warning if passed to a components style prop
+const omittedStyleProperties = [
+  'errorBorder',
+  'placeholderTextColor',
+  'selectionColor',
+  'underlineColorAndroid',
+  'withBorder',
+  'withoutBorder',
+];
 
 class TextInput extends PureComponent {
-  render() {
-    const { props } = this;
-    const style = {
-      ...props.style,
+  constructor(props) {
+    super(props);
+
+    autoBindReact(this);
+
+    this.state = {
+      isFocused: props.highlightOnFocus && props.autoFocus,
     };
-    delete style.placeholderTextColor;
-    delete style.selectionColor;
-    delete style.underlineColorAndroid;
+  }
+
+  handleBlur() {
+    this.setState({ isFocused: false });
+  }
+
+  handleFocus() {
+    this.setState({ isFocused: true });
+  }
+
+  render() {
+    const { errorMessage, highlightOnFocus, style } = this.props;
+    const { isFocused } = this.state;
 
     return (
-      <RNTextInput
-        {...props}
-        style={style}
-        placeholderTextColor={props.style.placeholderTextColor}
-        selectionColor={props.style.selectionColor}
-        underlineColorAndroid={props.style.underlineColorAndroid}
-      />
+      <View>
+        <RNTextInput
+          {..._.omit(this.props, 'style')}
```

Comment:
this is usually done like this: `const { errorMessage, highlightOnFocus, style, ...otherProps } = this.props;` and then here you'd just say `{...otherProps}`.
I think this would be the correct approach here as we don't want to send `errorMessage` nor `highlightOnFocus` to the RNTextInput anyway.